### DISCUSSION TITLE
Desktop stream

### DISCUSSION THREAD

**Discussion by IamLupo** [alvr-org/ALVR#1084] [General]
I just wanted to give a idea for desktop stream support.
In airlink you can enable desktop and have it overlapping your virtual environment. I really loved this feature to keep track of my desktop while i play my vr games.

I know you guys are working hard on improving and support as much devices. But having this feature in mind would be really nice :D
Love so far your work! Keep it up!

**Comment by zmerp** [comment]
Desktop streaming was already proposed in #258. We decided that ALVR should not increase its scope. If you need to access the desktop you can use the SteamVR dashboard.

What you are describing is a very unique use-case, but I think it could be implemented as a SteamVR overlay.

**Reply by IamLupo** [reply]
Ow really thank you for pointing out SteamVR overlay,.. i never did look into it enough. Apperntly it had precisely what i needed.

**Comment by darvon123** [comment]
can you at least give us a recommendation to a free alternative to virtual desktop

**Reply by zmerp** [reply]
There is Immersed which is free (although what the company is doing with Visor is not really commendable)

**Reply by darvon123** [reply]
I don't own a stand an alone headset like a _Meta Quest 3_ to use with the [immersed](https://immersed.com/) app. the next option is [Virtual Desktop](https://store.steampowered.com/app/382110/Virtual_Desktop_Classic/) cost between $15 to $25 to use, so that's definitely off the table. That leave [Desktop+](https://store.steampowered.com/app/1494460/Desktop/) as the best option for me to use as it currently **free** to use with steam VR. I still don't get why no one told me about this option


## Side note

judging from your description and I might be wrong here, if this app is really relying on a **hypervisor** to well function it possible means it emulating several different devices likely in a (kernel-level virtual machine) **KVM** environment *circumventing any security protocol* to access the operating system. likely they're doing this stuff in software to get around Window's hardware policy that requires every device connected to the system to have its own address line, global unit identity key **(GUID)**, and registry entries to be allowed on the platform. but wait a minute doesn't Windows' offer its own Api library to work with virtual devices? yeah, I believe it's called **Hyper-V**. Okay I am very confused, what's wrong with using hypervisor?

---------
Okay I'm just now starting realized that you were talking about Visor the VR headset.

**Reply by darvon123** [reply]
I would like to thank you for the suggestion and apologize for not responding sooner
